EIPS/eip-2565.md

```diff
@@ -0,0 +1,90 @@
+
+This complexity formula was meant to approximate the difficulty of Karatsuba multiplication. However, we found a better approximation for modelling modular exponentiation. We recommend the following formula to better estimate the compuational complexity for varying input values:
+
+```
+def mult_complexity(x):
+    ceiling(x/64)^2
+```
+where is x is max(length_of_MODULUS, length_of_BASE). (x) is divided by 64 to account for the number of limbs in multiprecision arithmetic.
+
+You can find comparison of these two complexity fomulas for the current test vector as the following [spreadsheet](https://docs.google.com/spreadsheets/d/1Fq3d3wUjGN0R_FX-VPj7TKhCK33ac--P4QXB9MPQ8iw/edit?usp=sharing).
+
+### Option (3): Replace libraries used by ModExp precompiles
+ModExp benchmarks for different libraries can be found at the following [spreadsheet](https://docs.google.com/spreadsheets/d/1Fq3d3wUjGN0R_FX-VPj7TKhCK33ac--P4QXB9MPQ8iw/edit?usp=sharing).
+
+While alternative libraries can provide improved performance, this option is not recommended at this time.
+
+## Rationale
+
+### **Recommended** Option (1): Change value of GQUADDIVISOR:
+Changing the value of this parameter from 20 to 200 will reduce the gas cost of this precompile by a factor of 10 with minimal implementation changes. With this change, the cost of the ModExp precompile will have a higher cost (gas/second) than other precompiles such as ECRecover.
+
+![Option 1 Graph](../assets/eip-tbd/GQuad_Change.png)
+
+### Option (2): Modify ‘computational complexity’ formula
+An alternative ‘complexity’ function can be found at the following [spreadsheet](https://docs.google.com/spreadsheets/d/1Fq3d3wUjGN0R_FX-VPj7TKhCK33ac--P4QXB9MPQ8iw/edit?usp=sharing).
+
+![Option 2 Graph](../assets/eip-tbd/Complexity_Regression.png?raw=true)
+
+The new complexity function has a better fit vs. the execution time when compared to the current complexity function. This better fit is because the new complexity formula accounts for the use of binary exponentiation algorithms that are used by ‘bigint’ libraries for large exponents. You may also notice the regression line of the proposed complexity function bisects the test vector data points. This is because the run time varies depending on if the modulus is even or odd.
+
+While modifying the computational complexity formula can improve gas estimation at a medium implementation cost, we do not recommend it at this time.
+
+### Option (3): Improving the ModExp precompile implementations
+
+![Option 3 Graph](../assets/eip-tbd/Library_Benchmarks.png?raw=true)
+
+Replacing the underlying library can improve the performance of the ModExp precompile by 2x-4x for large exponents, but comes at a high implementation cost. We do not recommend this option at this time.
+
+## Test Cases
+As no underlying algorithms are being changed, there are no additional test cases to specify.
+
+## References
+[EIP-198](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-198.md?raw=true)
```

Comment:
```suggestion
[EIP-198](https://eips.ethereum.org/EIPS/eip-198)
```

Though not sure this References section is needed because EIP-198 is referenced a few times already.